src/sor.ts

```diff
@@ -40,17 +68,26 @@ export async function getSorSwap(chainId: number, order: Order): Promise<SwapInf
     pools
   );
 
+  const tokens = await getTokens();
+  console.log("All tokens: ", tokens);
+
   const { sellToken, buyToken, orderKind, amount } = order;
 
+  const sellTokenDetails: Token = await getToken(chainId, sellToken);
+  const buyTokenDetails: Token = await getToken(chainId, buyToken);
+
+  sor.swapCostCalculator.setNativeAssetPriceInToken(sellToken, sellTokenDetails.price);
```

Comment:
@johngrantuk does it make sense to set the price for both sell and buy tokens? Or does only one matter?